src/file.cpp

```diff
@@ -1111,6 +1149,30 @@ namespace libtorrent
 		remove(f, ec);
 	}
 
+	void remove_empty(std::string const& f, error_code& ec)
+	{
+		ec.clear();
+
+		file_status s;
+		stat_file(f, &s, ec);
+		if (ec) return;
+
+		if (s.mode & file_status::directory)
+		{
+			for (directory i(f, ec); !i.done(); i.next(ec))
+			{
+				if (ec) return;
+
+				std::string p = i.file();
+				if (p == "." || p == "..") continue;
+				remove_empty(combine_path(f, p), ec);
+				ec.clear();
+			}
+
+			remove(f, ec);
```

Comment:
I was responding to your comment: "Honestly, I could not think that there may be empty directories, not belonging to the torrent".
I would imagine that would be quite likely, in the scenario/use case you want to support